UPGRADE-GUIDE.md

```diff
@@ -0,0 +1,460 @@
+# Upgrade Guide
+
+This document provides a full guide for upgrading a Cosmos SDK chain from `v0.50.x` to `v0.53.x`.
+
+After completing this guide, applications will have:
+
+- The `x/protocolpool` module
+- The `x/epochs` module
+- Unordered Transaction support
+
+## Table of Contents
+
+- [App Wiring Changes](#app-wiring-changes)
+- [Adding ProtocolPool Module](#adding-protocolpool-module)
+    - [Manual Wiring](#protocolpool-manual-wiring)
+    - [DI Wiring](#protocolpool-di-wiring)
+- [Adding Epochs Module](#adding-epochs-module)
+    - [Manual Wiring](#epochs-manual-wiring)
+    - [DI Wiring](#epochs-di-wiring)
+- [Enable Unordered Transactions](#enable-unordered-transactions)
+- [Upgrade Handler](#upgrade-handler)
+
+## App Wiring Changes
+
+The `x/auth` module now contains a `PreBlocker` that _must_ be set in the module manager's `SetOrderPreBlockers` method.
```

Comment:
Can we flag this one as like SUPER NECESSARY

Reply:
marked all sections with REQUIRED or OPTIONAL